sdk/keyvault/azure-keyvault-keys/tests/test_challenge_auth_async.py

```diff
@@ -349,3 +351,75 @@ def verify(request):
     # ensure the mock sans I/O policies were called
     assert adder.on_request.called, "mock policy wasn't invoked"
     assert verifier.on_request.called, "mock policy wasn't invoked"
+
+
+@pytest.mark.asyncio
+@empty_challenge_cache
+@pytest.mark.parametrize("verify_challenge_resource", [True, False])
+async def test_verify_challenge_resource_matches(verify_challenge_resource):
+    """The auth policy should raise if the challenge resource doesn't match the request URL unless check is disabled"""
+
+    url = get_random_url()
+    token = "**"
+    resource = "https://bad-resource.azure.net"
```

Comment:
Would be a good idea to add a test - for a future regression test - to make sure your check against ".{domain}" e.g., ".vault.azure.net" continues working (note the preceeding ".").

Reply:
I just checked .NET's tests to understand what you mean by this, but I don't see any tests using a resource including a preceding ".", like "https://.vault.azure.net".

Do you mean that such a resource should be considered valid, or that it should be invalid even if "https://vault.azure.net" (without the preceding ".") would be correct? Or do you mean that we should still correctly identify a challenge resource as invalid if it's preceded by a "."?

Reply:
Oh, I think I understand what you mean. Are you saying that we should have a check that explicitly ensures a KV endpoint (`https://vault-name.vault.azure.net`) succeeds when the resource is `https://vault.azure.net`?

If so, I'm happy to add another test if you think it would be helpful. Otherwise, my thinking was that this regression is implicitly checked by the other tests in this file using URLs from `get_random_url` -- which all have a domain ending with `.vault.azure.net` -- and requiring a matching challenge resource. `get_random_url` was updated to make this work, since it didn't actually have the correct domain suffix before

Reply:
No, I'm saying a URI like "https://myvault.azure.net" should not succeed. This is why that preceding "." e.g., ".vault.azure.net", is important.

Reply:
Ahhh I see, that makes total sense. I'll get that added 🙂